src/Request.php

```diff
@@ -397,7 +397,7 @@ public function encodeData()
 			case static::ENCODING_JSON:
 				return json_encode($this->data);
 			case static::ENCODING_QUERY:
-				return http_build_query($this->data);
+				return (!is_null($this->data) ? http_build_query($this->data) : '');
```

Comment:
What about moving this null check to the start of this method and return an empty string for every encoding if `$data === null`?

Reply:
For raw, it would make sense. However for json, I expect null to encode as the string "null".

Reply:
Good point.